pkg/cache/cluster_info.go

```diff
@@ -711,3 +720,29 @@ func (m *ClusterInfo) processRemovedApplication(event *cacheevent.RemovedApplica
 			fmt.Sprintf("Application %s Removed", event.ApplicationID))
 	}
 }
+
+func (m *ClusterInfo) UpdateSchedulerConfig(conf *configs.SchedulerConfig) error {
+	rmID := ""
+	for _, pi := range m.partitions {
+		rmID = pi.RmID
+		break
+	}
+	updatedPartitions, deletedPartitions, err := m.applyConfigChanges(conf, rmID)
+	if err != nil {
+		return err
+	}
+	if len(updatedPartitions) > 0 {
+		result := m.sendUpdatedPartitionsToScheduler(updatedPartitions)
+		if !result.Succeeded {
+			return fmt.Errorf(result.Reason)
+		}
+	}
+	if len(deletedPartitions) > 0 {
+		result2 := m.sendDeletedPartitionsToScheduler(deletedPartitions)
+		if !result2.Succeeded {
+			return fmt.Errorf(result2.Reason)
+		}
+	}
+	return nil
+}
+
```

Comment:
It is still possible that failure might happen while trying to delete some partitions.
The `updatedPartition` is already processed, but `deletedPartition` is not, the config becomes inconsistent.
I am not sure if we can have a more robust way to handle this based on the current structure, maybe we can leave this as the future work. As the auto-refresh code is exactly the same.

Reply:
Yes, this can be an issue. I suggest to think about it in a follow up task, because this is how it works the auto-refresh as well.